specs/stack.spec.js

```diff
@@ -0,0 +1,144 @@
+            var expected = [1, 2, 3];
+            var stack = new Stack(size);
+            stack.push(...data);
+            stack.push(...additionData);
+            var result = stack.pop(stack.size());
+
+            assert.deepEqual(result, expected, `Should be in stack after exceeding of size ${expected}, but actual ${result}.`); 
+        });
+    });
+
+    describe('pop', function() {
+        it('should return null if stack is empty', function() {
+            var data = 1;
+            var size = 1;
+            var stack = new Stack(size);
+            stack.push(data);
+            var result = stack.pop(size);
+
+            assert(result === data, `Should return ${data}, but actual ${result}.`);
+        });
+
+        it('should return n last element of stack', function() {
+            var size = 4;
+            var data = [1, 2, 3, 4];
+            var stack = new Stack(size);
+            stack.push(...data);
+            var result = stack.pop(size);
+
+            assert(result === data, `Should return ${data}, but actual ${result}.`);
+        });
+
+        it('should return null if stack is empty', function() {
+            var stack = new Stack();
+            var result = stack.pop();
+
+            assert(result === null, `Should return null, but actual ${result}.`);
+        });
+
+        it('should delete elemets from stack after pop', function() {
+            var size = 4;
+            var data = [1, 2, 3, 4];
+            var expected = [1, 2];
+            var stack = new Stack(size);
+            stack.push(...data);
+            stack.pop(size / 2);
+            result = stack.pop(size / 2);
+
+            assert(result === expected, `Should be in stack after pop - ${expected}, but actual ${result}.`);
+        });
+    });
+
+    describe('merge', function() {
+        it('should merge added stack first if flag is true', function() {
+            var size = 2;
+            var data1 = [1, 2];
+            var data2 = [2, 3];
+            var expected = [2, 3, 1, 2];
+            var stack1 = new Stack(size);
+            stack1.push(...data1);
+            var stack2 = new Stack(size);
+            stack2.push(...data2);
+
+            var result = stack1.merge(stack2, true).pop(size * 2);
+
+            assert.deepEqual(result, expected, `Should return ${expected}, but actual ${result}.`)
+        });
+
+        it('should merge added stack last if flag is false', function() {
+            var size = 2;
+            var data1 = [1, 2];
+            var data2 = [2, 3];
+            var expected = [1, 2, 3, 4];
+            var stack1 = new Stack(size);
+            stack1.push(...data1);
+            var stack2 = new Stack(size);
+            stack2.push(...data2);
+            
+            var result = stack1.merge(stack2, true).pop(size * 2);
+
+            assert.deepEqual(result, expected, `Should return ${expected}, but actual ${result}.`)
+        });
+    });
+});
```

Comment:
The same problem in all tests 
`assert(result === element, `Should return ${element}, but actual ${result}.`);` is not correct. Please refer to slides of lecture for correct assertions.

Reply:
also add negative scenarios